Fix Travis CI and Windows builds

addaleax opened the pull request:
Use GCC 4.8 for Travis builds so that `-std=c++11` is accepted as a command line option.

addaleax committed:
Fix Travis CI build

Use GCC 4.8 for Travis builds so that `-std=c++11` is accepted
as a command line option, and update the used Node.js versions
in general.

Also, update the package.json entry for the `lzma-native`
dependency to ^1.0.3.

addaleax commented:
ping @kevva – this has been open for a while now? :/

mastilver commented:
@kevva I would love to see that merged in as it fixes windows build as well (see #4)

addaleax commented:
@kevva Ping again, you seem to be more active again now?

addaleax committed:
Remove isXz dependency

Since lzma-native already includes a function to test
for the .xz magic bytes, this is superfluous.

addaleax commented:
Updated again.
